firmware/esp32/gdoor/main.cpp

```diff
@@ -70,25 +75,39 @@ void setup() {
     Serial.setTimeout(1);
     Serial.println("GDOOR Setup start");
     GDOOR::setup(PIN_TX, PIN_TX_EN, PIN_RX);
+    WIFI_HELPER::setup();
+    MQTT_HELPER::setup(WIFI_HELPER::mqtt_server(), WIFI_HELPER::mqtt_port(), WIFI_HELPER::mqtt_topic_bus_tx());
+
+    mqtt_topic_bus_rx = WIFI_HELPER::mqtt_topic_bus_rx();
+    debug = WIFI_HELPER::debug();
+
     Serial.println("GDOOR Setup done");
 }
 
 void loop() {
-    while(1) {
-        GDOOR::loop();
-        GDOOR_DATA* rx_data = GDOOR::read();
-        if(rx_data != NULL) {
-            GDOOR_DATA_PROTOCOL busmessage = GDOOR_DATA_PROTOCOL(rx_data);
-            outputSerial(busmessage);
-        } else if (!GDOOR::active()) { // Neither RX nor TX active,
-            if (Serial.available() > 0) { // let's check the serial port if something is in buffer
-                String serialstr = Serial.readString();
-                serialstr.trim();
+    WIFI_HELPER::loop();
+    MQTT_HELPER::loop();
+    GDOOR::loop();
+    GDOOR_DATA* rx_data = GDOOR::read();
+    if(rx_data != NULL) {
+        GDOOR_DATA_PROTOCOL busmessage = GDOOR_DATA_PROTOCOL(rx_data);
+        output(busmessage, mqtt_topic_bus_rx);        
+    } else if (!GDOOR::active()) { // Neither RX nor TX active,
+        String str_received("");
+        if (Serial.available() > 0) { // let's check the serial port if something is in buffer
+            str_received = Serial.readString();
+            str_received.trim();    
+        } else {
+            str_received = MQTT_HELPER::receive();
+        }
 
-                if(!parseSerialCommand(serialstr)) { //Check if received string is a command
-                    GDOOR::send(serialstr); // Send to bus if it is not a command
-                }
+        if(str_received.length() > 0) {
+            if(!parse(str_received)) { //Check if received string is a command
+                GDOOR::send(str_received); // Send to bus if it is not a command
+                Serial.print("Send: ");
```

Comment:
These two lines could be removed again, right? Just noticed that it pollutes the serial output.
The other logs like `Serial.println("GDOOR Setup start")` are useful I think and not a big issue as they only appear once when booting.

Reply:
Or wrap it with the debug mode?